go/vt/orchestrator/logic/tablet_discovery.go

```diff
@@ -157,6 +158,31 @@ func refreshTabletsInCell(ctx context.Context, cell string, loader func(instance
 	refreshTablets(tablets, query, args, loader, forceRefresh)
 }
 
+// forceRefreshAllTabletsInShard is used to refresh all the tablet's information (both MySQL information and topo records)
+// for a given shard. This function is meant to be called before or after a cluster-wide operation that we know will
+// change the replication information for the entire cluster drastically enough to warrant a full forceful refresh
+func forceRefreshAllTabletsInShard(ctx context.Context, keyspace, shard string) {
+	log.Infof("force refresh of all tablets in shard - %v/%v", keyspace, shard)
+	if ctx == context.Background() {
+		refreshCtx, refreshCancel := context.WithTimeout(context.Background(), *topo.RemoteOperationTimeout)
+		defer refreshCancel()
+		ctx = refreshCtx
+	}
+	refreshTabletsInKeyspaceShard(ctx, keyspace, shard, func(instanceKey *inst.InstanceKey) {
+		DiscoverInstance(*instanceKey, true)
+	}, true)
+}
+
+// refreshTabletInfoOfShard only refreshes the tablet records from the topo-server for all the tablets
+// of the given keyspace-shard.
+func refreshTabletInfoOfShard(ctx context.Context, keyspace, shard string) {
+	log.Infof("refresh of tablet records of shard - %v/%v", keyspace, shard)
+	refreshTabletsInKeyspaceShard(ctx, keyspace, shard, func(instanceKey *inst.InstanceKey) {
```

Comment:
we don't need context.background here?

Reply:
No, this happens before the recovery starts. We should be using the ctx passed in.